nvflare/private/fed/client/client_executor.py

```diff
@@ -24,138 +24,20 @@
-    def __init__(self, uid, startup) -> None:
-        """To init the ClientExecutor.
-
-        Args:
-            uid: client name
-            startup: startup folder
-        """
-        pipe_path = startup + "/comm"
-        if not os.path.exists(pipe_path):
-            os.makedirs(pipe_path)
-
-        self.pipe = FilePipe(root_path=pipe_path, name="training")
-        self.logger = logging.getLogger(self.__class__.__name__)
-
-    def start_train(
-        self,
-        client,
-        job_id,
-        args,
-        app_root,
-        app_custom_folder,
-        listen_port,
-        allocated_resource,
-        token,
-        resource_consumer,
-        resource_manager,
-        target: str,
-    ):
-        """start_train method to start the FL client training.
-
-        Args:
-            client: the FL client object
-            job_id: the job_id
-            args: admin command arguments for starting the FL client training
-            app_root: the root folder of the running APP
-            app_custom_folder: FL application custom folder
-            listen_port: port to listen the command.
-            allocated_resource: allocated resources
-            token: token from resource manager
-            resource_consumer: resource consumer
-            resource_manager: resource manager
-            target: SP target location
-
-        """
-        pass
-
-    def check_status(self, client, job_id) -> str:
-        """To check the status of the running client.
-
-        Args:
-            client: the FL client object
-            job_id: the job_id
-
-        Returns:
-            A client status message
-        """
-        pass
-
-    def abort_train(self, client, job_id):
-        """To abort the client training.
-
-        Args:
-            client: the FL client object
-            job_id: the job_id
-        """
-        pass
-
-    def abort_task(self, client, job_id):
-        """To abort the client executing task.
-
-        Args:
-            client: the FL client object
-            job_id: the job_id
-        """
-        pass
-
-    def get_run_info(self, job_id) -> dict:
-        """Get the run information.
-
-        Args:
-            job_id: the job_id
-
-        Returns:
-            A dict of run information.
-        """
-        pass
-
-    def get_errors(self, job_id):
-        """Get the error information.
-
-        Returns:
-            A dict of error information.
-
-        """
-        pass
-
-    def reset_errors(self, job_id):
-        """Reset the error information."""
-        pass
-
-    def send_aux_command(self, shareable: Shareable, job_id):
-        """To send the aux command to child process.
-
-        Args:
-            shareable: aux message Shareable
-            job_id: the job_id
-        """
-        pass
-
-    def cleanup(self):
-        """Cleanup."""
-        self.pipe.clear()
-
-
-class ProcessExecutor(ClientExecutor):
+class ProcessExecutor:
```

Comment:
We used to have another implementation ThreadExecutor, which implement the ClientExecutor APIs using multi-threads, instead of separate sub-process. Probably still keep this API architecture with more options?